api/authorizations.ts

```diff
@@ -219,4 +186,287 @@ export function registerAuthApi(app: Express) {
 			res.status(500).json({Message: err.message});
 		}
 	});
+
+
+	type AuthReqParams = {id_unit: number, req: string, date: Date, scipers: number[], cas: string[], room_ids: number[]};
+	app.post("/api/auth_req",
+		checkAPICall(
+			{
+				authorize: (req) => req.user.canEditAuthorizations,
+				required: {
+					req (req) { return req.query.req; },
+					date (req) { return req.query.date; },
+					id_unit (req) { return req.query.id_unit; },
+					room_ids (req) { return req.query.room_ids },
+					scipers (req) { return req.query.scipers },
+					cas (req) { return req.query.cas },
+				},
+				validate: {
+					req: reqRegexp,
+					date: Date,
+					id_unit: Number,
+					room_ids: validateCommaSeparatedNumbers,
+					scipers: validateCommaSeparatedNumbers,
+					cas: validateCASList
+				}
+			}),
+		async (req: Request<AuthReqParams>, res) => {
+			const args = {
+				id_unit: req.params.id_unit,
+				authorization: req.params.req,
+				expiration_date: req.params.date.toLocaleDateString("en-GB"),
+				status: "Active",
+				type: "Chemical",
+				cas: req.params.cas.map(c => {
+					return {name: c, status: "New"};
+				}),
+				holders: req.params.scipers.map(sc => {
+					return {sciper: sc, status: "New"};
+				}),
+				rooms: req.params.room_ids.map(r => {
+					return {id: r, status: "New"};
+				}),
+			}
+			await createAuthorization(args, args.id_unit, req.prisma);
+			res.json({Message: "Ok"});
+		}
+	);
+
+	type AuthRenewParams = {req: string, date: Date};
+	app.post("/api/auth_renew",
+		checkAPICall(
+			{
+				authorize: (req) => req.user.canEditAuthorizations,
+				required: {
+					req (req) { return req.query.req; },
+					date (req) { return req.query.date; }
+				},
+				validate: {
+					req: reqRenewRegexp,
+					date: Date
+				}
+			}),
+		async (req: Request<AuthRenewParams>, res) => {
+			const reqParts = req.params.req.split("-");
+			const requestNumber = `${reqParts[0]}-${reqParts[1]}`;
+			const argsRenew = {
+				search: requestNumber,
+				type: "Chemical"
+			}
+
+			const auth = await getTheAuthorization(argsRenew, req.prisma);
+			const argsUpdate = {
+				expiration_date: (new Date(req.params.date)).toLocaleDateString("en-GB"),
+				status: "Active",
+				renewals: parseInt(reqParts[2])
+			};
+			await updateAuthorization(argsUpdate, auth, req.prisma)
+			res.json({Message: "Ok"});
+		}
+	);
+
+	type AddChemParams = {cas: string, en: string, auth: 'yes' | 'no' | 1 | 0, fr?: string};
+	app.post("/api/add_chem",
+		checkAPICall(
+			{
+				authorize: (req) => req.user.canEditChemicals,
+				required: {
+					cas (req) { return req.query.cas; },
+					en (req) { return req.query.en; },
+					auth (req) { return req.query.auth; }
+				},
+				validate: {
+					cas: singleCAS,
+					en: textRegexp,
+					fr: textRegexp,
+					auth: new RegExp("yes|no|1|0"),
```

Comment:
Please make a function that returns a Boolean (and change the type for this field in `AddChemParams` accordingly).